tools/scoring/score/dimensions/result.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Outputs of the configuration scoring tool."""
+from dataclasses import dataclass
+
+
+@dataclass
+class Result:
+  """
+    Container for floating-point results which
+    quantify success in each dimension
+
+    Attributes:
+      correct_virtual: Number of correct attempts within virtual devices
+      correct_reporting: Number of correct attempts within reporting devices
+      correct_ceiling_virtual: Number of correct attempts possible
+        within virtual devices
+      correct_ceiling_reporting: Number of correct attempts possible
+        within reporting devices
```

Comment:
not sure I understand the difference between the `ceiling` ones and non ceiling.

Reply:
`correct_{category}` is the number of attempts achieved whereas `correct_ceiling_{category}` is the number of attempts it is _possible_ to achieve. It's used in `result_{category}()` to normalize correct versus incorrect attempts.

Reply:
what do we mean by it is `possible` to achieve? can you please give an example?

Reply:
"The test has four questions. It is possible to get four questions correct. The ceiling is four."

It's a little confusing because in the context of `correct`, "attempt" implies action taken. However, it's possible to get something wrong because no attempt was made.

"The ceiling is four. You attempted three and got two right. Despite only one incorrect guess, two were wrong: one by action and one by default."